### PR TITLE
implement an empty http endpoint

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: adds an empty HTTP endpoint.

### CODE REVIEW

File: api/pkg/handler/routes_v1.go [hunk 1]
```diff
@@ -272,6 +276,28 @@ func (r Routing) listDigitaloceanSizes() http.Handler {
...
+// swagger:route GET /api/v1/openstack/sizes openstack listOpenstackSizes
+//
+// Lists sizes from openstack
+//
+//     Produces:
+//     - application/json
+//
+//     Responses:
+//       default: errorResponse
+//       200: []OpenstackSize
```
Comment (p0lyn0mial) [inline api/pkg/handler/routes_v1.go:288]: I think this is correct.

File: api/pkg/handler/request.go [hunk 2]
```diff
@@ -150,6 +150,29 @@ func decodeDoSizesReq(c context.Context, r *http.Request) (interface{}, error) {
...
+	AuthURL    string
+	Username   string
+	Password   string
+	Tenant     string
+	Domain     string
+	Region     string
+}
+
+func decodeOpenstackSizesReq(c context.Context, r *http.Request) (interface{}, error) {
+	var req OpenstackSizesReq
```
Comment (p0lyn0mial) [inline api/pkg/handler/request.go]: holds the list of data that are required to make the request. 

At the moment they come from HTTP Headers but we could also read them from body (which would be a little more convenient when calling an API manually)

File: api/pkg/provider/cloud/openstack/provider.go [hunk 3]
```diff
@@ -19,18 +19,20 @@ const (
...
-type openstack struct {
+// Openstack TODO: add desc
+type Openstack struct {
 	dcs map[string]provider.DatacenterMeta
 }
 
 // NewCloudProvider creates a new digitalocean provider.
-func NewCloudProvider(dcs map[string]provider.DatacenterMeta) provider.CloudProvider {
-	return &openstack{
+func NewCloudProvider(dcs map[string]provider.DatacenterMeta) *Openstack {
```
Comment (p0lyn0mial) [inline api/pkg/provider/cloud/openstack/provider.go]: I'm planning to add one more method to `Openstack` and I didn't want to add this method to `provider.CloudProvider` thus I decided to return a pointer to `Openstack` struct.

If we don't like it there are alternatives ways of doing this.

**Comment by p0lyn0mial** [comment]
@mrIncompetent this pull is the next one to be reviewed.

File: api/pkg/handler/routes_v1.go [hunk 4]
```diff
@@ -88,6 +88,10 @@ func (r Routing) RegisterV1(mux *mux.Router) {
 		Path("/digitalocean/sizes").
 		Handler(r.listDigitaloceanSizes())
 
+	mux.Methods(http.MethodGet).
+		Path("/openstack/sizes").
```
Comment (mrIncompetent) [inline api/pkg/handler/routes_v1.go:92]: Not sure if we should call it sizes or flavors.

**Review by mrIncompetent — APPROVED** [review]

File: api/pkg/handler/routes_v1.go (hunk 4 above)
Comment (p0lyn0mial) [reply api/pkg/handler/routes_v1.go:92]: It aligns with `DO` naming conventions. In addition it is nice to have a common language that describes the same thing regardless of the underlying provider.

**Pull request merged**